crates/router/src/core/routing/helpers.rs

```diff
@@ -543,3 +551,350 @@ pub fn get_default_config_key(
+            &payment_attempt.profile_id,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::ProfileNotFound {
+            id: payment_attempt.profile_id.get_string_repr().to_owned(),
+        })?;
+
+    let default_success_based_routing_configs =
+        fetch_success_based_routing_configs(state, &business_profile)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("unable to retrieve default success_rate configs")?;
+
+    let tenant_business_profile_id = format!(
+        "{}:{}",
+        state.tenant.redis_key_prefix,
+        business_profile.get_id().get_string_repr()
+    );
+
+    let success_based_connectors = client
+        .calculate_success_rate(
+            tenant_business_profile_id.clone(),
+            default_success_based_routing_configs.clone(),
+            routable_connectors.clone(),
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("unable to calculate/fetch success rate from dynamic routing service")?;
+
+    let payment_status_attribute =
+        get_desired_payment_status_for_success_routing_metrics(&payment_attempt.status);
+
+    let first_success_based_connector_label = &success_based_connectors
+        .labels_with_score
+        .first()
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable(
+            "unable to derive first connector from the list of success rate connectors",
+        )?
+        .label
+        .to_string();
+
+    let first_success_based_connector = first_success_based_connector_label
+        .split_once(':')
+        .map(|(connector, _)| connector)
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable(
+            "unable to split connector_name and mca_id from success rate connector",
+        )?;
+
+    let outcome = get_success_based_metrics_outcome_for_payment(
+        &payment_status_attribute,
+        payment_connector.to_string(),
+        first_success_based_connector.to_string(),
+    );
+
+    core_metrics::DYNAMIC_SUCCESS_BASED_ROUTING.add(
+        &metrics::CONTEXT,
+        1,
+        &add_attributes([
+            ("tenant", state.tenant.name.clone()),
+            (
+                "merchant_id",
+                payment_attempt.merchant_id.get_string_repr().to_string(),
+            ),
+            (
+                "profile_id",
+                payment_attempt.profile_id.get_string_repr().to_string(),
+            ),
+            (
+                "merchant_connector_id",
+                payment_attempt
+                    .merchant_connector_id
+                    .as_ref()
+                    .ok_or(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("mca_id not found in payment_attempt")?
+                    .get_string_repr()
+                    .to_string(),
```

Comment:
Can't we obtain mca_id from the above response of dynamic routing service itself 

```
first_success_based_connector_label
        .split_once(':')
        .map(|(connector, mca_id)| connector)
```